scripts/vtk_convert.py

```diff
@@ -0,0 +1,91 @@
+from vtk import vtkPolyDataReader, vtkStructuredPointsReader, \
+    vtkXMLImageDataWriter, vtkXMLPolyDataWriter
+from girder_client import GirderClient
+import os
+import click
+import re
+
+@click.command()
+@click.option('--inputfolderid', required=True, prompt=True, help='Source Folder ID')
+@click.option('--outputfolderid', required=True, prompt=True, help='Destination Folder ID')
+@click.option('--apikey', required=True, prompt=True, help='API Key from Girder')
+@click.option('--hostname', default='https://data.computational-biology.org/api/v1/', prompt=True, help='URL, make sure to include /api/v1/ at the end')
+def main(inputfolderid, outputfolderid, apikey, hostname):
+  gc = GirderClient(apiUrl=hostname)
+  gc.authenticate(apiKey=apikey)
+
+  def getFiles(itemData):
+    print('getting ' + itemData['name'])
+    dataFiles = gc.get('item/' + itemData['_id'] + '/files')
+    return dataFiles[0]
+
+  def getItems(folderData):
+    folderInfo = gc.get('folder/' + folderData['_id'] + '/details')
+    folderItems = gc.get('item',
+      parameters={
+        'folderId': folderData['_id'],
+        'limit': folderInfo['nItems']
+      })
+    return map(getFiles, folderItems)
+
+  gmsFolders = gc.get('folder',
+    parameters={
+      'parentType': 'folder',
+      'parentId': inputfolderid,
+    })
+
+  gmsFileNames = map(getItems, gmsFolders)
+
+  for row in gmsFileNames:
+    for elem in row:
+      oldName = elem['name']
+      print('dowloading ' + oldName)
+      try:
+        gc.downloadFile(elem['_id'], os.path.abspath(oldName))
+
+        reader = vtkPolyDataReader()
+        writer = vtkXMLPolyDataWriter()
+        fileType = ''
+        if re.search("^geometry_", oldName):
+          reader = vtkStructuredPointsReader()
+          reader.ReadAllScalarsOn()
+          writer = vtkXMLImageDataWriter()
+          fileType = 'geometry_'
+        elif re.search("^spore_", oldName):
+          fileType = 'spore_'
+        elif re.search("^macrophage_", oldName):
+          fileType = 'macrophage_'
+        else:
+          raise Exception('Invalid File: {}'.format(oldName))
+
+        reader.SetFileName(oldName)
+        reader.Update()
+        data = reader.GetOutput()
+
+        end = oldName[len(fileType):]
+        num = '00' + end[:len(end) - 4]
+        fixed = num[len(num) - 3:]
+        name = fileType + fixed + '.vti'
+        print('writing ' + name)
+
+        writer.SetFileName(name)
+        writer.SetCompressorTypeToZLib()
+        writer.SetInputData(data)
+        writer.Update()
+
+        timepointFolder = gc.post('folder/',
+          parameters={
+            'parentType': 'folder',
+            'parentId': outputfolderid,
+            'name': fixed,
+            'reuseExisting': True
+          })
+
+        gc.uploadFileToFolder(timepointFolder['_id'], name)
+
+      except girder_client.HttpError:
```

Comment:
If a request fails, we first want to retry it (since maybe it was a transient network failure). See this: https://stackoverflow.com/questions/15431044/can-i-set-max-retries-for-requests-request for how to retry with Python requests.

Note, we don't want to instantiate our own `Session` in this case; we will use the one that `GirderClient` provides us with: https://github.com/girder/girder/blob/master/clients/python/girder_client/__init__.py#L240

Finally, if a request keeps failing after some number of retries, I think we should error out and abort the script. We don't want the script to complete and report a success status even when some of the files within it (maybe at a point where the user doesn't see the error message) didn't convert properly. The web app needs to be able to assume that all of the files exist and are working properly.